Progress x
Progress y
Progress size
Progress skew
bool goodenough

